oap-server/server-receiver-plugin/envoy-metrics-receiver-plugin/src/main/java/org/apache/skywalking/oap/server/receiver/envoy/als/LogEntry2MetricsAdapter.java

```diff
@@ -81,10 +81,16 @@ public HTTPServiceMeshMetric.Builder adaptToUpstreamMetrics() {
         final long outboundStartTime = startTime + formatAsLong(properties.getTimeToFirstUpstreamTxByte());
         final long outboundEndTime = startTime + formatAsLong(properties.getTimeToLastUpstreamRxByte());
 
+        final HTTPServiceMeshMetric.Builder builder = adaptCommonPart();
+        // For client side call, status needs to be overridden because 4xx http status codes
+        // are considered as errors to.
+        final boolean status = builder.getResponseCode() < 400;
+
         return adaptCommonPart()
```

Comment:
Can we share the two `adaptCommonPart()`s?

```suggestion
        final HTTPServiceMeshMetric.Builder builder = adaptCommonPart();
        // For client side call, status needs to be overridden because 4xx http status codes
        // are considered as errors to.
        final boolean status = builder.getResponseCode() < 400;

        return builder
```